packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/CredentialsImpl.kt

```diff
@@ -56,25 +54,30 @@ internal class CredentialsImpl constructor(
             RealmInterop.realm_app_credentials_new_apple(Validation.checkEmpty(idToken, "idToken"))
 
         internal fun facebook(accessToken: String): RealmCredentialsPointer =
-            RealmInterop.realm_app_credentials_new_facebook(Validation.checkEmpty(accessToken, "accessToken"))
+            RealmInterop.realm_app_credentials_new_facebook(
+                Validation.checkEmpty(
+                    accessToken,
+                    "accessToken"
+                )
+            )
 
         internal fun google(token: String, type: GoogleAuthType): RealmCredentialsPointer {
             Validation.checkEmpty(token, "token")
             return when (type) {
-                GoogleAuthType.AUTH_CODE -> RealmInterop.realm_app_credentials_new_google_auth_code(token)
-                GoogleAuthType.ID_TOKEN -> RealmInterop.realm_app_credentials_new_google_id_token(token)
+                GoogleAuthType.AUTH_CODE -> RealmInterop.realm_app_credentials_new_google_auth_code(
+                    token
+                )
+                GoogleAuthType.ID_TOKEN -> RealmInterop.realm_app_credentials_new_google_id_token(
+                    token
+                )
             }
         }
 
         internal fun jwt(jwtToken: String): RealmCredentialsPointer =
             RealmInterop.realm_app_credentials_new_jwt(Validation.checkEmpty(jwtToken, "jwtToken"))
 
-        internal fun customFunction(payload: Any): RealmCredentialsPointer =
-            BsonEncoder.encodeToBsonValue(payload).let { bsonValue: BsonValue ->
-                require(bsonValue.bsonType == BsonType.DOCUMENT) {
-                    "Invalid payload type '${payload::class.simpleName}', only BsonDocument and maps are supported."
-                }
-                RealmInterop.realm_app_credentials_new_custom_function(Bson.toJson(bsonValue))
-            }
+        @PublishedApi
+        internal fun customFunction(payload: String): RealmCredentialsPointer =
```

Comment:
Maybe rename payload to something that indicate the type? I assume this is a ejson string so maybe `ejsonPayload` ?